internal/providers/terraform/parser.go

```diff
@@ -630,7 +632,7 @@ func addressModulePart(addr string) string {
 
 	if len(ap) >= 3 && ap[len(ap)-3] == "data" {
 		mp = ap[:len(ap)-3]
-	} else {
+	} else if len(ap) >= 2 {
 		mp = ap[:len(ap)-2]
 	}
```

Comment:
```suggestion
```suggestion
	}
	
	if len(ap) >= 2 {
		mp = ap[:len(ap)-2]
	}
```